services/OpenAIService.py

```
@@ -356,3 +356,68 @@ def health_check(self) -> bool:
+                },
+                tokens_used=response.usage.total_tokens,
+                model_name=response.model,
+                created_at=datetime.fromtimestamp(response.created, UTC),
+            )
+
+        except Exception as e:
+            logger.error(f"Error analyzing activity schema with OpenAI: {str(e)}")
+            if isinstance(e, RoboRateLimitError):
+                raise
+            if "rate limit" in str(e).lower():
+                raise RoboRateLimitError("OpenAI rate limit exceeded")
```

Ruff (0.8.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)